crates/typst-library/src/model/par.rs

```diff
@@ -108,6 +108,132 @@ pub struct ParElem {
+    ///       })
+    ///
+    ///     kinds.split("spacing").map(x => x.len() + 1)
+    ///   }
+    ///
+    ///   let jumps = n-lines
+    ///     .map(n => ((text-height,) * n).intersperse(par.leading))
+    ///     .intersperse(par.spacing)
+    ///     .flatten()
+    ///
+    ///   place(grid(
+    ///     ..jumps
+    ///       .enumerate()
+    ///       .map(((i, h)) => if calc.even(i) {
+    ///         // Draw a stripe for the line
+    ///         block(
+    ///           height: h,
+    ///           width: 100%,
+    ///           fill: c.par-line,
+    ///         )
+    ///       } else {
+    ///         // Put an annotation for the gap
+    ///
+    ///         let sw(value-for-leading, value-for-spacing) = if h == par.leading {
+    ///           value-for-leading
+    ///         } else {
+    ///           value-for-spacing
+    ///         }
+    ///
+    ///         align(end, block(
+    ///           height: h,
+    ///           outset: (right: sw(0.5em, 1em)),
+    ///           stroke: (
+    ///             left: none,
+    ///             rest: 0.5pt + sw(c.leading-line, c.spacing-line),
+    ///           ),
+    ///           if i / 2 <= sw(n-lines.first(), n-lines.slice(0, 2).sum()) {
+    ///             place(horizon, dx: 1.3em, {
+    ///               set text(0.8em, sw(c.leading-text, c.spacing-text))
+    ///               sw([leading], [spacing])
+    ///             })
+    ///           },
+    ///         ))
+    ///       })
+    ///   ))
+    ///
+    ///   // Mark top and bottom edges
+    ///   place(
+    ///     // pos: top/bottom edge
+    ///     // dy: Δy to the last mark
+    ///     // kind: leading/spacing
+    ///     for (pos, dy, kind) in (
+    ///       (bottom, text-height, "leading"),
+    ///       (top, par.leading, "leading"),
+    ///       (
+    ///         bottom,
+    ///         (n-lines.first() - 1) * line-height - par.leading,
+    ///         "spacing",
+    ///       ),
+    ///       (top, par.spacing, "spacing"),
+    ///     ) {
+    ///       v(dy)
+    ///
+    ///       let c-text = c.at(kind + "-text")
+    ///       let c-line = c.at(kind + "-line")
+    ///
+    ///       place(end, box(
+    ///         height: 0pt,
+    ///         grid(
+    ///           columns: 2,
+    ///           column-gutter: 0.2em,
+    ///           align: pos,
+    ///           move(
+    ///             // Compensate optical illusion
+    ///             dy: if pos == top { -0.2em } else { 0.05em },
+    ///             text(0.8em, c-text)[#repr(pos) edge],
+    ///           ),
+    ///           line(length: 1em, stroke: 0.5pt + c-line),
+    ///         ),
+    ///       ))
+    ///     },
+    ///   )
+    /// }
+    ///
+    /// // Bury metadata for the main text
+    /// #set par.line(numbering: n => [#metadata(n) <par-line>])
+    ///
+    /// #set par(justify: true)
+    /// #set text(luma(50%))
```

Comment:
I don't like that the hyphen/comma sticks out of the colored area.

```suggestion
    /// #set text(luma(50%), overhang: false)
```

Also why luma is used? Does this increase contrast? I think it makes the text washed and less readable.

Reply:
I use luma because I want the reader to focus on annotations in the margin. The lorem text is not the point, and it's not meant be read.